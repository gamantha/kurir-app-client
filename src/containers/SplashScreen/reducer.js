export const ALREADY_LAUNCH = 'src/containers/splash-screen/already-launch';

export const launched = () => ({
    type: ALREADY_LAUNCH
});

const initialState = {
    loading: false,
    alreadyLaunch: false
};

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case ALREADY_LAUNCH:
            return { ...state, alreadyLaunch: true };
        default:
            return state;
    }
};

export default reducer;
