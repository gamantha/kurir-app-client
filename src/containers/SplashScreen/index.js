/* eslint-disable */
import React, { Component } from 'react';
import {
    Animated,
    Image,
    ActivityIndicator,
    Easing,
    AsyncStorage
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import jwtDecode from 'jwt-decode';
import Toast from 'react-native-simple-toast';
import { createStructuredSelector } from 'reselect';

import { images } from '../../assets';
import Api from '../../services/api';
import { reqRefreshToken } from '../UserLogin/reducer';

class SplashScreen extends Component {
    componentDidMount() {
        const { navigate } = this.props.navigation;
        let accessToken;
        AsyncStorage.getItem('accessToken')
            .then(res => {
                if (res === null || typeof res === 'undefined') {
                    navigate('Onboard');
                } else {
                    accessToken = res;
                    Api.setAuthorizationToken(accessToken);
                    return AsyncStorage.getItem('refreshToken');
                }
            })
            .then(refreshToken => {
                if (accessToken && accessToken.length > 1) {
                    const { exp } = jwtDecode(accessToken);
                    if (exp < Math.floor(Date.now() / 1000) - 300) {
                        this.props.refreshToken(refreshToken);
                    } else {
                        navigate('Main');
                    }
                }
            })
            .catch(err => Toast.show(err.message));
    }

    render() {
        return (
            <Animated.View>
                <Image
                    style={{ width: '100%', height: '100%' }}
                    source={images.splash}
                />
            </Animated.View>
        );
    }
}

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            refreshToken: reqRefreshToken
        },
        dispatch
    );

export default connect(null, mapDispatchToProps)(SplashScreen);
