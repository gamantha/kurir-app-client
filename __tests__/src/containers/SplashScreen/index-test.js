import React from 'react';
import SplashScreen from '../../../../src/containers/SplashScreen';
import Toast from 'react-native-simple-toast';
import renderer from 'react-test-renderer';

jest.mock('react-native-simple-toast', () => 'Toast');
test('SplashScreen renders correctly', () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
