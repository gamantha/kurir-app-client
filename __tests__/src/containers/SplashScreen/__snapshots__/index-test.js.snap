// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders correctly 1`] = `
<View
  collapsable={undefined}
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/splash.png",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
